### PR TITLE
Restore webview state after rotation

### LINKED ISSUE #1169: HABPanel is reset when device orientation is changed
### Actual behaviour
HABPanel's starting page is shown after the device orientation is changed.

### Expected behaviour
HABPanel should be on the same subpage as before.

### Can you reproduce the issue in demo mode?

Yes

#### Client
* App version : 2.5.6-beta

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #1169

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/activity/WebViewFragment.kt [hunk 1]
```diff
@@ -163,7 +162,11 @@ class WebViewFragment : Fragment(), ConnectionFactory.UpdateListener {
 
     fun goBack(): Boolean {
         if (webView?.canGoBack() == true) {
+            val oldUrl = webView?.url
             webView?.goBack()
+            if (oldUrl == webView?.url) {
+                goBack()
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/activity/WebViewFragment.kt]: Isn't this equivalent to returning `webView?.url != oldUrl`? I'd find this more elegant than the recursive call.

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/activity/WebViewFragment.kt]: No, it's not the same. If `goBack()` returns `false`, HABPanel is closed.
I added this code to "filter out" redundant entries in the back stack.

Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/activity/WebViewFragment.kt]: I see. The recursion still irks me a bit. How about this:
```
val oldUrl = webView?.url
do {
    webView?.goBack()
} while (webView?. url == oldUrl && webView?.canGoBack() == true)
```

And maybe add a comment like `// skip redundant history entries while going back`

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/activity/WebViewFragment.kt]: Picked.

**Pull request merged**